### PR TITLE
fix: compliance change id and title naming

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>

## Description
compliance change id and title naming

## Related issues
- Related #3315
- Related #3130

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).

### CODE REVIEW

File: docs/docs/cloud/aws/compliance.md [hunk 1]
```diff
@@ -336,13 +336,13 @@ Summary Report for compliance: awscis1.2
...
-$ trivy aws --compliance=awscis1.2 --report=summary --format=json
+$ trivy aws --compliance=aws-cis-1.2 --report=summary --format=json
 ```
 
 ```json
 {
-	"ID": "0001",
-	"Title": "awscis1.2",
+ "ID": "aws-cis-1.2",
+ "Title": "AWS CIS Foundations",
```
Comment (knqyf263) [inline docs/docs/cloud/aws/compliance.md:345]: mistake? Spaces are removed.

File: pkg/types/report.go [hunk 2]
```diff
@@ -51,10 +51,10 @@ const (
 	ClassCustom      = "custom"
 
 	// ComplianceNsa is the compliance checks for nsa
-	ComplianceNsa      = Compliance("nsa")
-	ComplianceCIS      = Compliance("cis")
-	ComplianceAWSCIS12 = Compliance("awscis1.2")
-	ComplianceAWSCIS14 = Compliance("awscis1.4")
+	ComplianceNsa      = Compliance("k8s-nsa-1.0")
+	ComplianceCIS      = Compliance("k8s-cis-1.23")
```
Comment (knqyf263) [inline pkg/types/report.go:55]:
```suggestion
	ComplianceK8sNSA      = Compliance("k8s-nsa-1.0")
	ComplianceK8sCIS      = Compliance("k8s-cis-1.23")
```

**Review by knqyf263 — COMMENTED** [review]
Tests are failing. Everything else looks good. I left nitpicks.

File: pkg/types/report.go [hunk 3]
```diff
@@ -51,10 +51,10 @@ const (
 	ClassCustom      = "custom"
 
 	// ComplianceNsa is the compliance checks for nsa
-	ComplianceNsa      = Compliance("nsa")
-	ComplianceCIS      = Compliance("cis")
-	ComplianceAWSCIS12 = Compliance("awscis1.2")
-	ComplianceAWSCIS14 = Compliance("awscis1.4")
+	ComplianceNsa      = Compliance("k8s-nsa-1.0")
```
Comment (knqyf263) [inline pkg/types/report.go:54]: Isn't it 1.2? Does it mean we don't support 1.2, but 1.0?
https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF

```suggestion
	ComplianceNsa      = Compliance("k8s-nsa-1.2")
```

Comment (chen-keinan) [reply pkg/types/report.go:54]: when we implemented (in starboard) and now its for 1.0 , there are minor changes between 1.2 to 1.0 we could revise it to 1.2 , maybe we can do it later

Comment (chen-keinan) [reply pkg/types/report.go:54]: note they are not keeping only latest spec

Comment (knqyf263) [reply pkg/types/report.go]: Do we want to keep all the minor versions?

- k8s-nsa-1.0
- k8s-nsa-1.1
- k8s-nsa-1.2
- (k8s-nsa-1.3)

If no, what if we change it to `k8s-nsa`?

Comment (chen-keinan) [reply pkg/types/report.go]: sound good to keep it `k8s-nsa` in cli but in docs and defSec we should specified that its 1.0 until we update it

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**